LayoutTests/fast/forms/textarea-scrolled-endline-caret.html

```diff
@@ -13,18 +14,22 @@
     ta.blur();
 
     ta.focus();
+
     // click
     if (window.eventSender) {
         testRunner.waitUntilDone();
         requestAnimationFrame(() => {
-            eventSender.mouseMoveTo(90, 20);
-            eventSender.mouseDown();
-            eventSender.mouseUp();
-            if (ta.selectionEnd == 17)
-                res.innerHTML = "Test Succeeded";
-            else
-                res.innerHTML = "Test Failed: caret is at " + ta.selectionEnd;
-            testRunner.notifyDone();
+            
+            setTimeout(function() {
+                eventSender.mouseMoveTo(90, 20);
+                eventSender.mouseDown();
+                eventSender.mouseUp();
+                if (ta.selectionEnd == 17)
+                    res.innerHTML = "Test Succeeded";
+                else
+                    res.innerHTML = "Test Failed: caret is at " + ta.selectionEnd;
+                testRunner.notifyDone();
+            }, 0);
```

Comment:
It seems what you're waiting for here is the next rendering update?

There's a helper method, `UIHelper.renderingUpdate()`, in `ui-helper.js` that you might find useful here:

```
UIHelper.renderingUpdate().then(() => {
    eventSender.mouseMoveTo(90, 20);
    eventSender.mouseDown();
    eventSender.mouseUp();
    if (ta.selectionEnd == 17)
        res.innerHTML = "Test Succeeded";
    else
        res.innerHTML = "Test Failed: caret is at " + ta.selectionEnd;
    testRunner.notifyDone();
});
```